subsys/bluetooth/host/hci_core.h

```diff
@@ -381,6 +381,12 @@ struct bt_dev {
 	/* Local Identity Resolving Key */
 	uint8_t			irk[CONFIG_BT_ID_MAX][16];
 
+	/* Only 1 RPA per identity */
+	bt_addr_le_t		rpa[CONFIG_BT_ID_MAX];
+
+	/* RPA valid flag per identity */
+	atomic_t		rpa_valid[CONFIG_BT_ID_MAX];
```

Comment:
This seems like it’s wasting 31 bits of memory per  identity, since you only have one flag. I think `ATOMIC_DEFINE(rpa_valid, CONFIG_BT_ID_MAX)` would make more sense.

Reply:
Another alternative, which would save even more memory, would be to use `BT_ADDR_LE_NONE` to indicate that the RPA for a given identity is not valid. No separate flags needed then.

Reply:
used the BT_ADDR_LE_NONE approach. Removed the flags